src/init.cpp

```diff
@@ -1900,6 +1911,23 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
 
     connOptions.m_i2p_accept_incoming = args.GetBoolArg("-i2pacceptincoming", DEFAULT_I2P_ACCEPT_INCOMING);
 
+    if (args.GetBoolArg("-privatebroadcast", DEFAULT_PRIVATE_BROADCAST)) {
+        // If -listenonion is set, then NET_ONION may not be reachable now
+        // but may become reachable later, thus only error here if it is not
+        // reachable and will not become reachable for sure.
+        if (!g_reachable_nets.Contains(NET_I2P) && !g_reachable_nets.Contains(NET_ONION) &&
+            (!listenonion || args.IsArgSet("-onlynet"))) {
```

Comment:
ok so then isn't it possible for a user to start with `-privatebroadcast` but i2p and onion aren't reachable *yet*? They would get an init error but the Tor daemon could become reachable later, by the time we need it.

Reply:
It would be possible.
`-privatebroadcast=1 -listenonion=1 -torcontrol=... -torpassword=...` would do exactly that and allow the startup to proceed. However, `args.IsArgSet("-onlynet")` was misplaced. It was intended to give error on things like
`-privatebroadcast=1 -listenonion=1 -torcontrol=... -torpassword=... -onlynet=ipv4` but it was too strong and would mistakenly forbid
`-privatebroadcast=1 -listenonion=1 -torcontrol=... -torpassword=... -onlynet=onion` as well.

Changed.